Adds word counts to fields in bulk seo page and sidebar panel

Mesuva opened the pull request:
Uses the jQuery Text Counter Plugin under MIT
https://github.com/ractoon/jQuery-Text-Counter

<img width="698" alt="screen shot 2015-11-02 at 9 40 20 pm" src="https://cloud.githubusercontent.com/assets/1079600/10880341/5f65ccce-81aa-11e5-9cc9-69d5d7cecabf.png">

<img width="1071" alt="screen shot 2015-11-02 at 9 42 40 pm" src="https://cloud.githubusercontent.com/assets/1079600/10880375/aa039a90-81aa-11e5-96e4-35a56e080b68.png">

Mesuva commented:
I've tried to follow the logic as to how third party libraries should be included for grunt, but apologies if I'm not 100% correct!

aembler commented:
This looks busy to me, honestly. What do other people think? Could this be handled with a browser extension?

Mesuva commented:
I did try out the use of badges/labels to try to make it look a bit neater, but I struggled to come up something better, so I just left it as fairly boring markup. I agree that it does look busy. 

I thought it would still be better for me to suggest its inclusion, even when it's not pretty, as it is pretty useful for SEO tasks.

aembler commented:
Yeah, I'm wondering if maybe keeping it as part of the bulk SEO updater only? I'm not sure. It'd probably be fine regardless. What if they were displayed using the text-muted class and floated to the right above the text field?

MrKarlDilkington commented:
+1

I think having the count in the bulk SEO updater and Page Settings SEO would be very useful.

katzueno commented:
+1

I like it.

I think this page will be used by those of who are taking SEO seriously. And they want to know if their page title will be included by Google or not.

joe-meyer commented:
+1 for the character counter. I can't tell you how many times I open another program to copy and paste text into to do just that. I agree it could be placed to the right of the title / floated right though to conserve a bit more space.

aembler commented:
I think we just need to float those titles to the right and this would be ready to add. Looks good otherwise.

Mesuva commented:
Moving the text counter is actually proving a problem.

I'm able to move the counter in the DOM using the counter's init callback, but after that has happened it disconnects from actually counting the text field (it must coded to always update a div after the field).

Also applying more than one class to the counter container seems to break the counting (I tried something like 'label label-primary' to see how it would look).

I picked the counter script based on its flexibility and how recently it had been updated, but it obviously has some limitations.

MrKarlDilkington commented:
@Mesuva 

I believe you can make the divs with the class of .form-group relative by adding another class, then make .help-desk absolute with top -5px and right 0px.

Mesuva commented:
I've updated PR this as per MrKD's suggestions.

It now looks as per the attached.  (I adjusted the spacing/width of the bulk SEO page as it was a bit unusually constrained).

![screen shot 2016-01-05 at 9 47 57 pm](https://cloud.githubusercontent.com/assets/1079600/12113952/3f61d958-b3f6-11e5-9ed9-0d3d14bd8e1e.png)

![screen shot 2016-01-05 at 9 48 08 pm](https://cloud.githubusercontent.com/assets/1079600/12113958/4f0c438e-b3f6-11e5-8e90-692ebc27edcd.png)
